tests/check_check_sub.c

```diff
@@ -553,6 +553,95 @@ START_TEST(test_ck_assert_ptr_ne)
 }
 END_TEST
 
+START_TEST(test_ck_assert_mem_eq)
+{
+  const char *s = "\x00\x00\x00\x00\x02";
+  ck_assert_mem_eq("\x00\x00\x00\x00\x02", s, 5);
+  record_failure_line_num(__LINE__);
+  ck_assert_mem_eq("\x00\x00\x00\x00\x01", s, 5);
+}
+END_TEST
+
+START_TEST(test_ck_assert_mem_ne)
+{
+  const char *s = "\x00\x00\x00\x00\x02";
+  const char *t = "\x00\x00\x00\x00\x01";
+  ck_assert_mem_ne(t, s, 5);
+  t = "\x00\x00\x00\x00\x02";
+  record_failure_line_num(__LINE__);
+  ck_assert_mem_ne(t, s, 5);
+}
+END_TEST
+
+START_TEST(test_ck_assert_mem_lt)
+{
+  const char *s = "\x00\x00\x00\x00\x01";
+  const char *t = "\x00\x00\x00\x00\x02";
+  ck_assert_mem_lt(s, t, 5);
+  record_failure_line_num(__LINE__);
+  ck_assert_mem_lt(s, s, 5);
+}
+END_TEST
+
+START_TEST(test_ck_assert_mem_le)
+{
+  const char *s = "\x00\x00\x00\x00\x01";
+  const char *t = "\x00\x00\x00\x00\x02";
+  ck_assert_mem_le(s, t, 5);
+  ck_assert_mem_le(s, s, 5);
+  record_failure_line_num(__LINE__);
+  ck_assert_mem_le(t, s, 5);
+}
+END_TEST
+
+START_TEST(test_ck_assert_mem_gt)
+{
+  const char *s = "\x00\x00\x00\x00\x01";
+  const char *t = "\x00\x00\x00\x00\x02";
+  ck_assert_mem_gt(t, s, 5);
+  record_failure_line_num(__LINE__);
+  ck_assert_mem_gt(t, t, 5);
+}
+END_TEST
+
+START_TEST(test_ck_assert_mem_ge)
+{
+  const char *s = "\x00\x00\x00\x00\x01";
+  const char *t = "\x00\x00\x00\x00\x02";
+  ck_assert_mem_ge(t, s, 5);
+  ck_assert_mem_ge(t, t, 5);
+  record_failure_line_num(__LINE__);
+  ck_assert_mem_ge(s, t, 5);
+}
+END_TEST
+
+START_TEST(test_ck_assert_mem_expr)
+{
+  const char *s = "\x00\x00\x00\x00\x01";
+  const char *t[] = { "\x00\x00\x00\x00\x01", "\x00\x00\x00\x00\x02" };
+  int i = -1;
+  ck_assert_str_eq(s, t[++i]);
+  ck_assert_str_eq(s, t[i]);
+}
+END_TEST
+
+START_TEST(test_ck_assert_mem_zerolen)
+{
+  const char *s = "\x00\x00\x00\x00\x02";
+  const char *t = "\x00\x00\x00\x00\x01";
+  ck_assert_mem_eq(t, s, 0);
+}
+END_TEST
+
+START_TEST(test_ck_assert_mem_eq_long)
+{
+  const char *s = "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02";
+  const char *t = "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01";
```

Comment:
Yes, but I need the different character at the end of the array, not at the beginning.